pokemon-showdown

```diff
@@ -33,14 +33,16 @@ try {
 } catch (e) {}
 
 if (config && config.golang) {
-	if (!process.env.GOPATH) {
-		console.log('The GOPATH environment variable is not set! It is required in order to run the server using Go.');
-		process.exit(0);
-	}
-	if (!process.env.GOROOT) {
-		console.log('The GOROOT environment variable is not set! It is required in order to run the server using Go.');
-		process.exit(0);
-	}
+	// GOPATH and GOROOT are optional to a degree, but we need them in order
+	// to be able to handle Go dependencies. Since Go only cares about the
+	// first path in the list, so will we.
+	const GOPATH = child_process.execSync('go env GOPATH', {stdio: null, encoding: 'utf8'})
+		.trim().split(path.delimiter)[0];
+	process.env.GOPATH = GOPATH;
+
+	let GOROOT = child_process.execSync('go env GOROOT', {stdio: null, encoding: 'utf8'})
```

Comment:
Again, no need to deal with GOROOT. In fact, you shouldn't set GOPATH either. Values from `go env` are values that Go knows, and they can be overridden by setting them as environment variables.

You want to retrieve GOPATH for purposes of moving files to that location, but that's about it.